tests/js/dictionary-tests.js

```diff
@@ -655,6 +655,60 @@ module.exports = {
         realm.write(() => realm.create(DictSchema.name, { a: { x: null } }));
         let data = realm.objects(DictSchema.name)[0].a;
         TestCase.assertEqual(data.x, null, "Should be an equals to mutable.x = null");
+        realm.close();
+    },
+
+    testDictionaryEmbeddedObject() {
+        const Child = {
+            name: "Children",
+            embedded: true,
+            properties: {
+                num: "int"
+            }
+        };
+
+        const DictTypedSchema = {
+            name: "TypedDictionary",
+            properties: {
+                dict1: "Children{}",
+                dict2: "Children{}"
+            }
+        };
+
+        const DictMixedSchema = {
+            name: "MixedDictionary",
+            properties: {
+                dict1: "{}",
+                dict2: "{}"
+            }
+        };
+
+        let realm = new Realm({schema: [DictTypedSchema, DictMixedSchema, Child]})
+        realm.write(() => {
+            realm.create(DictTypedSchema.name, {
+                dict1: {children1: { num: 2 }, children2: { num: 3 }},
+                dict2: {children1: { num: 4 }, children2: { num: 5 }}
+            });
+        });
+
+        realm.write(() => {
+            const error = new Error("Only Realm instances are supported.");
+            TestCase.assertThrows(() => {
+                realm.create(DictMixedSchema.name, {
+                    dict1: {children1: { num: 2 }, children2: { num: 3 }},
```

Comment:
Seems to me like you're not asserting anything about this object, is that an oversight?

Reply:
I am checking if an exception is thrown as an embedded object cannot be used as value for mixed.